### ISSUE TITLE
About loss function

### ISSUE THREAD

**Issue by Ckeai** [BorealisAI/de-simple#7]
Thank you very much for sharing. I have a question. I found that you used cross entropy loss in this code，but the second parameter of this loss is a vector filled with zeros. I don't understand this setting. I think that we shoould assign labels to positive and negative samples.  Could you tell me the reason for the setting of this loss? Thank you very much!

**Comment by RishabGoel** [comment]
Hi @Ckeai,
Thanks for reaching out. I am assuming that you are referring to line 54 in `trainer.py` file. The 0s in that array indicate the index in the softmax which is the positive example. The 0th element of the `scores_reshaped` variable is the positive tuple and the rest of the elements in the softmax are corresponding to negative samples.  The cross-entropy function requires the target variable to have the class index of the ground-truth label. The way we generate the batch (see `addNegFacts2` function in `dataset.py`) makes sure that the 0th index is the positive tuple. Please refer to the following link.
https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html

I hope this helps. Let me know if you were not referring to this.

**Comment by Ckeai** [comment]
I'm glad you answered my question. With your help, I completely understood this. Thank you very much.
